# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/FileInfoT.h
# Commit: 7e9d56eb485e
# Full Hash: 7e9d56eb485e06cca2a82fe88e2ae36c2ccd2095
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-04-01 21:26:59
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileManager. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65549
# ==============================================================================

diff -r a753c018a4a8 -r 7e9d56eb485e dom/indexedDB/FileInfoT.h
--- a/dom/indexedDB/FileInfoT.h	Wed Apr 01 09:54:24 2020 +0000
+++ b/dom/indexedDB/FileInfoT.h	Wed Apr 01 09:52:54 2020 +0000
@@ -9,6 +9,7 @@
 
 #include "nsISupportsImpl.h"
 #include "nsCOMPtr.h"
+#include "SafeRefPtr.h"
 
 namespace mozilla {
 namespace dom {
@@ -20,7 +21,7 @@
   using AutoLock = typename FileManager::AutoLock;
 
   FileInfoT(const typename FileManager::FileManagerGuard& aGuard,
-            RefPtr<FileManager> aFileManager, const int64_t aFileId,
+            SafeRefPtr<FileManager> aFileManager, const int64_t aFileId,
             const nsrefcnt aInitialDBRefCnt = 0);
 
   void AddRef();
@@ -30,7 +31,7 @@
 
   void GetReferences(int32_t* aRefCnt, int32_t* aDBRefCnt);
 
-  FileManager* Manager() const;
+  FileManager& Manager() const;
 
   int64_t Id() const;
 
@@ -49,7 +50,7 @@
   ThreadSafeAutoRefCnt mRefCnt;
   ThreadSafeAutoRefCnt mDBRefCnt;
 
-  const RefPtr<FileManager> mFileManager;
+  const SafeRefPtr<FileManager> mFileManager;
 };
 
 }  // namespace indexedDB